tiatoolbox/dataloader/wsireader.py

```diff
@@ -1111,17 +1134,18 @@ def read_rect(self, location, size, resolution=1.0, units="baseline"):
             units=units,
         )
 
-        im_region = self.img[
-            level_location[1] : level_location[1] + baseline_read_size[1],
-            level_location[0] : level_location[0] + baseline_read_size[0],
-            :,
-        ]
+        bounds = utils.transforms.locsize2bounds(
+            location=level_location, size=baseline_read_size
+        )
+        im_region = utils.image.safe_padded_read(
+            self.img, bounds, pad_mode="constant", constant_values=255
+        )
 
-        im_region = transforms.imresize(
+        im_region = utils.transforms.imresize(
             img=im_region, scale_factor=post_read_scale, output_size=size
         )
 
-        im_region = transforms.background_composite(image=im_region)
+        im_region = utils.transforms.background_composite(image=im_region)
         return im_region
 
     def read_bounds(self, bounds, resolution=1.0, units="baseline"):
```

Comment:
Add kwargs to the input so that padding kwargs can be pass in.

Reply:
I may do this in a separate PR to reduce the complexity of this one as it is already getting quite complex.